cmd/boostx/utils_cmd.go

```diff
@@ -0,0 +1,272 @@
+		if err != nil {
+			return err
+		}
+
+		n, err := clinode.Setup(ctx, sdir)
+		if err != nil {
+			return err
+		}
+
+		api, closer, err := lcli.GetGatewayAPI(cctx)
+		if err != nil {
+			return fmt.Errorf("cant setup gateway connection: %w", err)
+		}
+		defer closer()
+
+		walletAddr, err := n.Wallet.GetDefault()
+		if err != nil {
+			return err
+		}
+
+		log.Infow("selected wallet", "wallet", walletAddr)
+
+		ds := ds_sync.MutexWrap(datastore.NewMapDatastore())
+		messagesigner := messagesigner.NewMessageSigner(n.Wallet, &modules.MpoolNonceAPI{ChainModule: api, StateModule: api}, ds)
+
+		params, err := actors.SerializeParams(&walletAddr)
+		if err != nil {
+			return err
+		}
+
+		msg := &types.Message{
+			To:     marketactor.Address,
+			From:   walletAddr,
+			Value:  amt,
+			Method: marketactor.Methods.AddBalance,
+			Params: params,
+		}
+
+		head, err := api.ChainHead(ctx)
+		if err != nil {
+			return err
+		}
+		basefee := head.Blocks()[0].ParentBaseFee
+
+		spec := &lapi.MessageSendSpec{
+			MaxFee: abi.NewTokenAmount(1000000000), // 1 nFIL
+		}
+		msg, err = api.GasEstimateMessageGas(ctx, msg, spec, types.EmptyTSK)
+		if err != nil {
+			return xerrors.Errorf("GasEstimateMessageGas error: %w", err)
+		}
+
+		msg.GasFeeCap = big.Mul(big.Int(basefee), big.NewInt(2)) // use 2*basefee, so that this message confirms quickly
+
+		smsg, err := messagesigner.SignMessage(ctx, msg, func(*types.SignedMessage) error { return nil })
+		if err != nil {
+			return err
+		}
+
+		cid, err := api.MpoolPush(ctx, smsg)
+		if err != nil {
+			return xerrors.Errorf("mpool push: failed to push message: %w", err)
+		}
+		log.Infow("submitted market-add message", "cid", cid.String())
+
+		return nil
+	},
+}
+
+var commpCmd = &cli.Command{
+	Name:      "commp",
+	Usage:     "",
+	ArgsUsage: "<inputPath>",
+	Before:    before,
+	Action: func(cctx *cli.Context) error {
+		if cctx.Args().Len() != 1 {
+			return fmt.Errorf("usage: commP <inputPath>")
+		}
+
+		inPath := cctx.Args().Get(0)
+
+		rdr, err := os.Open(inPath)
+		if err != nil {
+			return err
+		}
+		defer rdr.Close() //nolint:errcheck
+
+		// check that the data is a car file; if it's not, retrieval won't work
+		_, err = car.ReadHeader(bufio.NewReader(rdr))
+		if err != nil {
+			return xerrors.Errorf("not a car file: %w", err)
+		}
+
+		if _, err := rdr.Seek(0, io.SeekStart); err != nil {
+			return xerrors.Errorf("seek to start: %w", err)
+		}
+
+		w := &writer.Writer{}
+		_, err = io.CopyBuffer(w, rdr, make([]byte, writer.CommPBuf))
+		if err != nil {
+			return xerrors.Errorf("copy into commp writer: %w", err)
+		}
+
+		commp, err := w.Sum()
+		if err != nil {
+			return xerrors.Errorf("computing commP failed: %w", err)
+		}
+
+		encoder := cidenc.Encoder{Base: multibase.MustNewEncoder(multibase.Base32)}
+
+		fmt.Println("CommP CID: ", encoder.Encode(commp.PieceCID))
+		fmt.Println("Piece size: ", types.NewInt(uint64(commp.PieceSize.Unpadded().Padded())))
+		return nil
```

Comment:
Suggest we also output the CAR file root here because they need it for the deal command

Reply:
I am doing that in the `generate-car` command.